src/shared/Core/PlatformUtils.cs

```diff
@@ -176,52 +177,109 @@ public static bool IsElevatedUser()
         {
             try
             {
                 if (IsWindows())
                 {
-                    return GetWindowsArgv0();
+                    return GetWindowsEntryPath();
                 }
 
                 if (IsMacOS())
                 {
-                    return GetMacOSArgv0();
+                    return GetMacOSEntryPath();
                 }
 
                 if (IsLinux())
                 {
-                    return GetLinuxArgv0();
+                    return GetLinuxEntryPath();
                 }
             }
             catch
             {
-                // If there are any issues getting the native argv[0]
+                // If there are any issues getting the native entry path
                 // we should not throw, and certainly not crash!
                 // Just return null instead.
             }
 
             return null;
         }
 
-        private static string GetLinuxArgv0()
+        private static string GetLinuxEntryPath()
         {
+            // Try to extract our native argv[0] from the original cmdline
             string cmdline = File.ReadAllText("/proc/self/cmdline");
-            return cmdline.Split('\0')[0];
+            string argv0 = cmdline.Split('\0')[0];
+
+            // argv[0] is an absolute file path
+            if (Path.IsPathRooted(argv0))
+            {
+                return argv0;
+            }
+
+            string path = Path.GetFullPath(
+                Path.Combine(Environment.CurrentDirectory, argv0)
+            );
+
+            // argv[0] is relative to current directory (./app) or a relative
+            // name resolved from the current directory (subdir/app).
+            // Note that we do NOT want to consider the case when it is just
+            // a simple filename (argv[0] == "app") because that would actually
+            // have been resolved from the $PATH instead (handled below)!
+            if ((argv0.StartsWith("./") || argv0.IndexOf('/') > 0) && File.Exists(path))
+            {
+                return path;
+            }
+
+            // argv[0] is a name that was resolved from the $PATH
+            string pathVar = Environment.GetEnvironmentVariable("PATH");
+            if (pathVar != null)
+            {
+                string[] paths = pathVar.Split(':');
+                foreach (string pathBase in paths)
+                {
+                    path = Path.Combine(pathBase, argv0);
+                    if (File.Exists(path))
+                    {
+                        return path;
+                    }
+                }
+            }
+
+            //
+            // We cannot determine the absolute file path from argv[0]
+            // (how we were launched), so let's now try to extract the
+            // fully resolved executable path from /proc/self/exe.
+            // Note that this means we may miss if we've been invoked
+            // via a symlink, but it's better than nothing at this point!
```

Comment:
I'll play devil's advocate here for a moment: why do we need the absolute path? Is the purpose of this functionality not to determine whether the program was called via the `git-credential-manager-core` symlink?

If that is the case, using `/proc/self/exe` won't give us what we need because it already points to the resolved path of the actual executable, not to the symlink.

And if that is the case, we actually do not need the absolute path at all, right? If we see that `argv[0]` is `git-credential-manager-core` (without any slash, i.e. not an absolute nor a relative path), we know that we want to warn, right?

Reply:
Okay, that makes sense. I do suspect, though, that we need to look at `argv[0]` instead of `/proc/self/exe` for that `-core` rename warning.

Reply:
Indeed we will. But since this is just a warning, and there may be some cases where `argv[0]` is missing/empty, we can fall-back on `/proc/self/exe` to get _a_ valid(*) path to the real executable for the more important `(un)configure` command usage.

(*) unless someone deletes us before we read this link of course 🤡